kiva/fonttools/_scanner.py

```diff
@@ -0,0 +1,538 @@
+# (C) Copyright 2005-2021 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+"""
+####### NOTE #######
+This is based heavily on matplotlib's font_manager.py rev 8713,
+but has been modified to not use other matplotlib modules
+####################
+
+Authors   : John Hunter <jdhunter@ace.bsd.uchicago.edu>
+            Paul Barrett <Barrett@STScI.Edu>
+            Michael Droettboom <mdroe@STScI.edu>
+Copyright : John Hunter (2004,2005), Paul Barrett (2004,2005)
+License   : matplotlib license (PSF compatible)
+            The font directory code is from ttfquery,
+            see license/LICENSE_TTFQUERY.
+"""
+import glob
+import logging
+import os
+import subprocess
+import sys
+
+from fontTools.ttLib import TTCollection, TTFont, TTLibError
+
+from kiva.fonttools import afm
+from kiva.fonttools._constants import weight_dict
+from kiva.fonttools._util import get_ttf_prop_dict, weight_as_number
+
+logger = logging.getLogger(__name__)
+# Error message when fonts fail to load
+_FONT_ENTRY_ERR_MSG = "Could not convert font to FontEntry for file %s"
+
+# OS Font paths
+MSFolders = r"Software\Microsoft\Windows\CurrentVersion\Explorer\Shell Folders"
+MSFontDirectories = [
+    r"SOFTWARE\Microsoft\Windows NT\CurrentVersion\Fonts",
+    r"SOFTWARE\Microsoft\Windows\CurrentVersion\Fonts",
+]
+OSXFontDirectories = [
+    "/Library/Fonts/",
+    "/Network/Library/Fonts/",
+    "/System/Library/Fonts/",
+]
+X11FontDirectories = [
+    # an old standard installation point
+    "/usr/X11R6/lib/X11/fonts/TTF/",
+    # here is the new standard location for fonts
+    "/usr/share/fonts/",
+    # documented as a good place to install new fonts
+    "/usr/local/share/fonts/",
+    # common application, not really useful
+    "/usr/lib/openoffice/share/fonts/truetype/",
+]
+
+home = os.environ.get("HOME")
+if home is not None:
+    # user fonts on OSX
+    path = os.path.join(home, "Library", "Fonts")
+    OSXFontDirectories.append(path)
+    path = os.path.join(home, ".fonts")
+    X11FontDirectories.append(path)
+
+
+class FontEntry(object):
+    """ A class for storing Font properties. It is used when populating
+    the font lookup dictionary.
+    """
+    def __init__(self, fname="", name="", style="normal", variant="normal",
```

Comment:
~not sure what the difference between `fname` and `name` is.~ looking at the rest of the class, it's clear that `fname` is the filepath and `name` is the font name. Would it be possible to rename `fname` to `filepath` or will that break backwards compatibility?

Reply:
🤷‍♂️ This one is probably OK. I don't think any external code is dealing directly with `FontEntry` instances.